Assets/VRTK/Scripts/Locomotion/VRTK_TouchpadMovement.cs

```diff
@@ -0,0 +1,480 @@
+        public float slideDeceleration = 0.1f;
+        [Tooltip("The delay before the next warp is allowed to happen.")]
+        public float warpDelay = 0.5f;
+        [Tooltip("The distance to warp in to the given direction.")]
+        public float warpRange = 1;
+        [Tooltip("The maximum altitude change allowed for a warp to happen.")]
+        public float warpMaxAltitudeChange = 1;
+
+        private GameObject controllerLeftHand;
+        private GameObject controllerRightHand;
+        private Transform playArea;
+        private Vector2 touchAxis;
+        private float movementSpeed = 0f;
+        private float strafeSpeed = 0f;
+        private float blinkFadeInTime = 0;
+        private float lastWarp = 0;
+        private float lastFlip = 0;
+        private float lastSnapRotate = 0;
+        private CapsuleCollider bodyCollider;
+        private Transform headset;
+        private bool leftSubscribed;
+        private bool rightSubscribed;
+        private ControllerInteractionEventHandler touchpadAxisChanged;
+        private ControllerInteractionEventHandler touchpadUntouched;
+
+        private void Awake()
+        {
+            touchpadAxisChanged = new ControllerInteractionEventHandler(DoTouchpadAxisChanged);
+            touchpadUntouched = new ControllerInteractionEventHandler(DoTouchpadTouchEnd);
+            playArea = VRTK_DeviceFinder.PlayAreaTransform();
+            controllerLeftHand = VRTK_DeviceFinder.GetControllerLeftHand();
+            controllerRightHand = VRTK_DeviceFinder.GetControllerRightHand();
+            if (!playArea)
+            {
+                Debug.LogError("No play area could be found. Have you selected a valid Boundaries SDK in the SDK Manager? If you are unsure, then click the GameObject with the `VRTK_SDKManager` script attached to it in Edit Mode and select a Boundaries SDK from the dropdown.");
+            }
+
+            headset = VRTK_DeviceFinder.HeadsetTransform();
+            if (!headset)
+            {
+                Debug.LogError("No headset could be found. Have you selected a valid Headset SDK in the SDK Manager? If you are unsure, then click the GameObject with the `VRTK_SDKManager` script attached to it in Edit Mode and select a Boundaries SDK from the dropdown.");
+            }
+        }
+
+        private void Start()
+        {
+            VRTK_PlayerObject.SetPlayerObject(gameObject, VRTK_PlayerObject.ObjectTypes.CameraRig);
+            SetControllerListeners(controllerLeftHand);
+            SetControllerListeners(controllerRightHand);
+
+            bodyCollider = playArea.GetComponent<CapsuleCollider>();
+            if (!bodyCollider)
+            {
+                Debug.LogError("No body collider could be found in the play area. VRTK_BodyPhysics script is required in one of the scene GameObjects.");
+            }
+        }
+
+        protected void OnAxisMovement(AxisMovementType movementType, AxisMovementDirection direction)
```

Comment:
Actually this is the only protected method, so it can be probably just moved up (or can it just be private?)